homeassistant/components/vallox/number.py

```diff
@@ -0,0 +1,152 @@
+
+from . import ValloxDataUpdateCoordinator, ValloxEntity
+from .const import DOMAIN
+
+
+class ValloxNumberEntity(ValloxEntity, NumberEntity):
+    """Representation of a Vallox number entity."""
+
+    entity_description: ValloxNumberEntityDescription
+    _attr_has_entity_name = True
+    _attr_entity_category: EntityCategory | None = EntityCategory.CONFIG
+
+    def __init__(
+        self,
+        name: str,
+        coordinator: ValloxDataUpdateCoordinator,
+        description: ValloxNumberEntityDescription,
+        client: Vallox,
+    ) -> None:
+        """Initialize the Vallox number entity."""
+        super().__init__(name, coordinator)
+
+        self.entity_description = description
+
+        self._attr_unique_id = f"{self._device_uuid}-{description.key}"
+        self._client = client
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the value reported by the sensor."""
+        if (
+            value := self.coordinator.data.get_metric(
+                self.entity_description.metric_key
+            )
+        ) is None:
+            return None
+
+        return float(value)
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Update the current value."""
+        await self._client.set_values(
+            {self.entity_description.metric_key: float(value)}
+        )
+        await self.coordinator.async_request_refresh()
+
+
+@dataclass
+class ValloxNumberEntityDescription(NumberEntityDescription):
+    """Describes Vallox number entity."""
+
+    metric_key: str = ""
+
+
+NUMBER_ENTITIES: tuple[ValloxNumberEntityDescription, ...] = (
+    ValloxNumberEntityDescription(
+        key="supply_air_target_home",
+        name="Supply air temperature (Home)",
+        metric_key="A_CYC_HOME_AIR_TEMP_TARGET",
+        device_class=NumberDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        icon="mdi:thermometer",
+        native_min_value=5.0,
+        native_max_value=25.0,
+        native_step=1.0,
+    ),
+    ValloxNumberEntityDescription(
+        key="supply_air_target_away",
+        name="Supply air temperature (Away)",
+        metric_key="A_CYC_AWAY_AIR_TEMP_TARGET",
+        device_class=NumberDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        icon="mdi:thermometer",
+        native_min_value=5.0,
+        native_max_value=25.0,
+        native_step=1.0,
+    ),
+    ValloxNumberEntityDescription(
+        key="supply_air_target_boost",
+        name="Supply air temperature (Boost)",
+        metric_key="A_CYC_BOOST_AIR_TEMP_TARGET",
+        device_class=NumberDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        icon="mdi:thermometer",
+        native_min_value=5.0,
+        native_max_value=25.0,
+        native_step=1.0,
+    ),
+    ValloxNumberEntityDescription(
+        key="fan_speed_home",
+        name="Fan speed (Home)",
```

Comment:
Couldn't the fan settings be different fan entities?

Reply:
Ventilation unit has a fan speed and temperature settings per each profile. If I understood your question.

Reply:
Why not create three fan entities instead of these three number entities?

Reply:
Ahh HomeAssistant has FanEntity. Did not know. Now I understand.
@MartinHjelmare If I extract fan entities, do you want it as a separate PR or I can add to current one?

Reply:
Please add it in a separate PR. Thanks!